server/routes/replayEventRouter.js

```diff
@@ -0,0 +1,57 @@
+const express = require('express');
+const replayEventRouter = express.Router();
+
+const { handleEvent, getDomainEvents, dispatch } = require('../models/DomainEvent');
+
+const Session = require('../infra/database/Session');
+const { publishMessage } = require('../infra/messaging/RabbitMQMessaging');
+
+const { EventRepository } = require('../infra/database/PgRepositories');
+const { handleVerifyCaptureProcessed } = require('../services/EventHandlers');
+
+replayEventRouter.post("/", async function(req, res) {
+    const session = new Session(false);
+    const eventRepository = new EventRepository(session);
+    const executeGetDomainEvents = getDomainEvents(eventRepository);
+    const domainEvents = await executeGetDomainEvents({"status":req.body.status});
+    if(req.body.status == "raised"){
+        const eventDispatch = dispatch(eventRepository, publishMessage);
+        try {
+            domainEvents.forEach(async (domainEvent) => {
+                const result = await executeGetDomainEvents({"id":domainEvent.id});
+                eventDispatch(domainEvent);
+            });
+        } catch(e) {
+            console.log(e);
+            if (session.isTransactionInProgress()){
+                await session.rollbackTransaction();
+            }
+            if (migrationSession.isTransactionInProgress()) {
+               await migrationSession.rollbackTransaction();
+            }
+            let result = e;
+            res.status(422).json({...result[0]});
+        }
+    }else if(req.body.status == "received"){
+        const eventHandle = handleEvent(handleVerifyCaptureProcessed);
+        domainEvents.forEach(async (domainEvent) => {
+            try {
+                eventHandle(domainEvent.payload);
+            } catch(e) {
+                console.log(e);
+                if (session.isTransactionInProgress()){

```

Comment:
We don't need transaction handling at this layer where event replay occurs. For events that are in `raised` status, all that we do in database is a single table update and it can occur without requiring transactions.  On the other hand, events in `received` status are to be handled involving business logic that might require more than one db table updates and would require transactions. But, that transactional code should be done by the event handlers and not here. For e.g. if you look at 
`handleVerifyCaptureProcessed` function in EventHandler.js,  it uses transaction to do the work.  Thus making it unnecessary to require transaction at this layer.